fvm/evm/stdlib/contract.cdc

```diff
@@ -288,6 +290,59 @@ contract EVM {
                 value: value.attoflow
             ) as! Result
         }
+
+        /// Bridges the given NFT to the EVM environment, requiring a Provider from which to withdraw a fee to fulfill
+        /// the bridge request
+        access(all)
+        fun depositNFT(
+            nft: @NonFungibleToken.NFT,
+            feeProvider: &{FungibleToken.Provider}
```

Comment:
nit: we may want to unify the feeprovider interface for all methods, 
Currently for some operations like withdraw tokens we don't collect any extra fee and we just charge the payer of the transaction.

Reply:
I'm not sure I follow. In this case, the `feeProvider` is providing a fee that is calculated by the bridge contracts. This could be a reference on a Vault, or, as currently implemented in the initial transactions, a `ScopedFTProvider` which enables the caller to set a withdrawal limit (see [this call](https://github.com/onflow/flow-evm-bridge/blob/main/cadence/transactions/bridge/nft/bridge_nft_to_evm.cdc#L80L83))

Reply:
I meant for flow tokens we don't collect any fees for bridging but we do charge for other fungible tokens, not 100% related to this PR, we can iterate over it later.